analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/OutputFileHandler.kt

```diff
@@ -0,0 +1,47 @@
+package de.maibornwolff.codecharta.serialization
+
+import java.io.BufferedWriter
+import java.io.File
+import java.io.FileWriter
+import java.io.OutputStreamWriter
+import java.io.PrintStream
+import java.io.Writer
+
+object OutputFileHandler {
+
+    fun checkAndFixFileExtension(outputName: String): String {
+        if (outputName.isEmpty()) return "default.cc.json"
+        val delimiter = extractDelimiter(outputName)
+        if (delimiter.isEmpty()) {
+            return outputName.substringBefore(".") + ".cc.json"
```

Comment:
This will not work if multiple dots are provided in the fileName like: "my.output.name". This would lead to my.cc.json which would not be correct. Instead you can use substringBeforeLast.

Reply:
@ce-bo We were considering using substringBeforeLast. If you are providing a file name such as "test.cc.json", this would lead to test.cc.cc.json which would also be incorrect. Therefore we decided that by convention fileNames do not involve dots.

Reply:
Restricting the filename would not be intuitive. So let's not do this. 
In case of "test.cc.json" we can simply check if "cc.json" is included or not. If yes, stop filename processing, if not, continue.